ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -200,7 +200,25 @@ public int compare(IndexEntry e1, IndexEntry e2) {
         // We mark as sealed all the index segments except the most recent index segment.
         // The recent index segment would go through recovery after they have been
         // read into memory
-        boolean sealed = i < indexFiles.size() - 1;
+        boolean sealed;
+        if (config.storeAutoCloseLastLogSegmentEnabled) {
+          //if journal.getFirstOffset() is null, it means active log segment has been created
+          //due to auto last log segment closed during compaction.And no entry has been added yet.
+          if (i == indexFiles.size() - 1) {
+            LogSegmentName logSegmentName =
+                IndexSegment.getIndexSegmentStartOffset(indexFiles.get(i).getName()).getName();
+            LogSegmentName lastLogSegmentName = log.getLastSegment().getName();
+            if (logSegmentName.compareTo(lastLogSegmentName) < 0) {
+              sealed = true;
+            } else {
+              sealed = false;
+            }
+          } else {
+            sealed = i < indexFiles.size() - 1;
```

Comment:
`i < indexFiles.size() - 1` is always True in this case, no?

Reply:
Yeah, Updated.